src/fsharp/UnicodeLexing.fs

```diff
@@ -7,22 +7,23 @@ module internal FSharp.Compiler.UnicodeLexing
 //
 
 open FSharp.Compiler.AbstractIL.Internal.Library
+open FSharp.Compiler.Features
 open Internal.Utilities
 open System.IO
 
 open Internal.Utilities.Text.Lexing
 
 type Lexbuf =  LexBuffer<char>
 
-let StringAsLexbuf (s:string) : Lexbuf =
-    LexBuffer<_>.FromChars (s.ToCharArray())
-  
-let FunctionAsLexbuf (bufferFiller: char[] * int * int -> int) : Lexbuf =
-    LexBuffer<_>.FromFunction bufferFiller 
+let StringAsLexbuf (supportsFeature: Features.LanguageFeature -> bool, s:string) : Lexbuf =
```

Comment:
@KevinRansom I know I spoke to you that I thought this approach was fine. Now that I'm looking at this more carefully, I don't think it is a good idea.

This feels like a code smell. I don't like that we need to pass `supportsFeature` in order to create a lexbuffer, it should not care about this. The only reason why is to be able to access `supportsFeature` in the parser (`.fsy`).
Lexbuffers and language features are two separate concerns and should not be intertwined this way; we have made this more complex.